src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassReportServiceImpl.java

```diff
@@ -0,0 +1,302 @@
+                .collect(Collectors.toList());
+        writeSummaryLineToReport(filteresdResults, resultCode, businessObjectClass);
+    }
+
+    private <R extends PersistableBusinessObject> void writeSummaryLineToReport(List<RassBusinessObjectUpdateResult<R>> filteresdResults,
+            RassObjectUpdateResultCode resultCode, Class businessObjectClass) {
+
+        String resultMessage = StringUtils.EMPTY;
+        switch (resultCode) {
+            case SUCCESS_NEW:
+                resultMessage = " created";
+                break;
+            case SUCCESS_EDIT:
+                resultMessage = " updated";
+                break;
+            case ERROR:
+                resultMessage = " that resulted in error";
+                break;
+            case SKIPPED:
+                resultMessage = " skipped";
+                break;
+        }
+
+        String reportMessage = "Number of " + businessObjectClass.getSimpleName() + " objects" + resultMessage + ": " + filteresdResults.size() + ".";
+        reportWriterService.writeFormattedMessageLine(reportMessage);
+    }
+
+    private void writeReportDetails(RassBatchJobReport rassBatchJobReport) {
+        reportWriterService.writeNewLines(1);
+        reportWriterService.writeSubTitle("*** Report Details ***");
+        reportWriterService.writeNewLines(1);
+        writeAgencyDetailsToReport(rassBatchJobReport);
+        reportWriterService.writeNewLines(1);
+        writeProposalDetailsToReport(rassBatchJobReport);
+        reportWriterService.writeNewLines(1);
+        writeAwardsDetailsToReport(rassBatchJobReport);
+        reportWriterService.writeNewLines(1);
+        reportWriterService.writeSubTitle("*** End of Report Details ***");
+        reportWriterService.writeNewLines(1);
+    }
+
+    private void writeAgencyDetailsToReport(RassBatchJobReport rassBatchJobReport) {
+        RassXmlProcessingResults processingResults = rassBatchJobReport.getProcessingResults();
+        RassBusinessObjectUpdateResultGrouping<Agency> agencyResultGrouping = processingResults.getAgencyResults();
+        List<RassBusinessObjectUpdateResult<Agency>> agencyResults = agencyResultGrouping.getObjectResults();
+        writeResultsDetailsToReport(agencyResults, Agency.class, true);
+    }
+
+    private void writeProposalDetailsToReport(RassBatchJobReport rassBatchJobReport) {
+        RassXmlProcessingResults processingResults = rassBatchJobReport.getProcessingResults();
+        RassBusinessObjectUpdateResultGrouping<Proposal> proposalResultGrouping = processingResults.getProposalResults();
+        List<RassBusinessObjectUpdateResult<Proposal>> proposalResults = proposalResultGrouping.getObjectResults();
+        writeResultsDetailsToReport(proposalResults, Proposal.class, false);
+    }
+
+    private void writeAwardsDetailsToReport(RassBatchJobReport rassBatchJobReport) {
```

Comment:
You may want to rename this method to "writeAwardDetailsToReport", for consistency with the naming schemes for the Agency and Proposal detail-writing methods above.

Reply:
updated